python/tskit/trees.py

```diff
@@ -3347,15 +3347,8 @@ def divergence(self, sample_sets, indices=None, windows=None, stat_type="site"):
             and end, so has one more entry than number of windows).
         :return: A list of the divergences, with diversities when i = j.
         """
-        if stat_type == "node":
-            if indices is not None:
-                raise ValueError("cannot specify more than one statistic with nodes")
-        if indices is None:
-            if len(sample_sets) != 2:
-                raise ValueError("Must have 2 sample sets.")
-            indices = [(0, 1)]
-
-        windows = parse_windows(windows, stat_type)
+        sample_sets, indices = self.check_input(sample_sets, indices, 2)
```

Comment:
The `check_input( )` function does this sort of thing. (although it doesn't do the "node" check, 'cause we don't have that implemented yet).